pkg/core/proxy/integrations/mysql/replayer/match.go

```
@@ -354,30 +355,58 @@ func matchClosePacket(_ context.Context, _ *zap.Logger, expected, actual mysql.P
 	return matchCount
 }
 
-func matchQueryPacket(_ context.Context, _ *zap.Logger, expected, actual mysql.PacketBundle) int {
-	matchCount := 0
+func getQueryStructure(sql string) (string, error) {
+	stmt, err := sqlparser.Parse(sql)
+	if err != nil {
+		return "", err // Propagate the parsing error
+	}
+
+	var structureParts []string
+	// Walk the AST and collect the Go type of each grammatical node.
+	_ = sqlparser.Walk(func(node sqlparser.SQLNode) (bool, error) {
```

Don't ignore this error while traversing through the query.